packages/dashboards/addon/services/dashboard-data.js

```diff
@@ -47,25 +50,33 @@ export default Ember.Service.extend({
    * @returns {Object} hash of widget id to data promise array
    */
   fetchDataForWidgets(dashboardId, widgets = [], decorators = [], options = {}) {
-    let result = {},
-      uuid = v1();
+    const result = {};
+    const uuid = v1();
 
     // Construct hash of widget id to data
     widgets.forEach(widget => {
-      let widgetId = get(widget, 'id'),
-        widgetDataPromises = get(widget, 'requests').map(request => {
-          //construct custom header for each widget with uuid
-          options.customHeaders = {
-            uiView: `dashboard.${dashboardId}.${uuid}.${widgetId}`
-          };
+      const widgetId = get(widget, 'id');
+      const dashboard = get(widget, 'dashboard');
+      const widgetDataPromises = get(widget, 'requests').map(request => {
+        //construct custom header for each widget with uuid
+        options.customHeaders = {
+          uiView: `dashboard.${dashboardId}.${uuid}.${widgetId}`
+        };
+
+        const requestWithFilters = this._applyFilters(dashboard, request);
+        const requestDecorated = this._decorate(decorators, requestWithFilters.serialize());
 
-          request = this._decorate(decorators, request.serialize());
+        const filterErrors = this._getFilterErrors(dashboard, request);
 
-          return this._fetch(request, options);
+        return this._fetch(requestDecorated, options).then(result => {
+          const serverErrors = getWithDefault(result, 'response.errors', []);
+
+          return merge({}, result, { response: { errors: [...serverErrors, ...filterErrors] } });
```

Comment:
I think we wanted this under the `meta` key under the response, any reason why we went with `errors`?

Reply:
I was following the JSON API errors object spec. https://jsonapi.org/examples/#error-objects-basics

Reply:
But these aren't errors, just information the user should know about the filters that may/may not been applied.